rp2040-hal/src/pwm/mod.rs

```diff
@@ -764,3 +769,135 @@ impl<S: SliceId, M: ValidSliceInputMode<S>> Slice<S, M> {
         pin.into().into_function()
     }
 }
+
+/// Type representing DMA access to PWM cc register.
+///
+/// Both channels are accessed together, because of narrow write replication.
+///
+/// ```no_run
+/// let buf = singleton!(: [CcFormat; 4] = [CcFormat{a: 0x1000, b: 0x9000}; 4]).unwrap();
+/// let buf2 = singleton!(: [CcFormat; 4] = [CcFormat{a: 0xf000, b: 0x5000}; 4]).unwrap();
+///
+/// let dma = pac.DMA.split(&mut pac.RESETS);
+///
+/// let dma_pwm = SliceDmaWrite::from(pwm);
+///
+/// let dma_conf = double_buffer::Config::new((dma.ch0, dma.ch1), buf, dma_pwm.cc);
+/// ```
+pub struct SliceDmaWriteCc<S: SliceId, M: ValidSliceMode<S>> {
+    slice: PhantomData<S>,
+    mode: PhantomData<M>,
+}
+
+/// Type representing DMA access to PWM top register.
+///
+/// ```no_run
+/// // Just set to something mesurable.
+/// pwm.channel_a.set_duty(0x1000);
+/// pwm.channel_b.set_duty(0x1000);
+///
+/// let buf = singleton!(: [u16; 4] = [0x7fff; 4]).unwrap();
+/// let buf2 = singleton!(: [u16; 4] = [0xffff; 4]).unwrap();
+///
+/// let dma = pac.DMA.split(&mut pac.RESETS);
+///
+/// // Reserve PWM slice for dma.
+/// let dma_pwm = SliceDmaWrite::from(pwm);
+///
+/// let dma_conf = double_buffer::Config::new((dma.ch0, dma.ch1), buf, dma_pwm.top);
+/// ```
+
+pub struct SliceDmaWriteTop<S: SliceId, M: ValidSliceMode<S>> {
+    slice: PhantomData<S>,
+    mode: PhantomData<M>,
+}
+
+/// PWM slice while used for DMA writes.
+/// ```no_run
+/// let dma_pwm = SliceDmaWrite::from(pwm);
+/// ```
+///
+pub struct SliceDmaWrite<S: SliceId, M: ValidSliceMode<S>> {
+    /// Part for top writes.
+    pub top: SliceDmaWriteTop<S, M>,
+
+    /// Part for cc writes.
+    pub cc: SliceDmaWriteCc<S, M>,
+    slice: Slice<S, M>,
+}
+
+impl<S: SliceId, M: ValidSliceMode<S>> From<Slice<S, M>> for SliceDmaWrite<S, M> {
+    fn from(value: Slice<S, M>) -> Self {
+        Self {
+            slice: value,
+            top: SliceDmaWriteTop {
+                slice: PhantomData::<S> {},
+                mode: PhantomData::<M> {},
+            },
+            cc: SliceDmaWriteCc {
+                slice: PhantomData::<S> {},
+                mode: PhantomData::<M> {},
+            },
```

Comment:
I think this can be simplified:
```suggestion
            top: SliceDmaWriteTop {
                slice: PhantomData,
                mode: PhantomData,
            },
            cc: SliceDmaWriteCc {
                slice: PhantomData,
                mode: PhantomData,
            },
```